Sming/Core/Data/Stream/SectionTemplate.h

```diff
@@ -0,0 +1,179 @@
+	   "{!pad:A:B:C} Copy of A padded to at least B characters with C (default is space). Use -ve B to left-pad. C")   \
+	XX(_repeat, "{!repeat:A:B} Repeat A, number of iterations is B")                                                   \
+	XX(_kb, "{!kb:A} Convert A to KB")                                                                                 \
+	XX(_ifdef, "{!ifdef:A}block{/if} emit block if A is not zero-length")                                              \
+	XX(_ifndef, "{!ifdef:A}block{/if} emit block if A is zero-length")                                                 \
+	XX(_ifeq, "{!ifeq:A:B} emit block if A == B")                                                                      \
+	XX(_ifneq, "{!ifneq:A:B} emit block if A != B")                                                                    \
+	XX(_ifgt, "{!ifgt:A:B} emit block if A > B")                                                                       \
+	XX(_iflt, "{!iflt:A:B} emit block if A < B")                                                                       \
+	XX(_ifge, "{!ifge:A:B} emit block if A >= B")                                                                      \
+	XX(_ifle, "{!ifle:A:B} emit block if A <= B")                                                                      \
+	XX(_ifbtw, "{!ifbtw:A:B:C} emit block if B <= A <= C")                                                             \
+	XX(_ifin, "{!ifin:A:B} emit block if A contains B")                                                                \
+	XX(_ifnin, "{!ifin:A:B} emit block if A does not contain B")                                                       \
+	XX(_else, "{!else}")                                                                                               \
+	XX(_endif, "{!endif}")                                                                                             \
+	XX(_add, "{!add:A:B} A - B")                                                                                       \
+	XX(_sub, "{!sub:A:B} A - B")                                                                                       \
+	XX(_goto, "{!goto:A} move to section A")                                                                           \
+	XX(_count, "{!count:A} emit number of records in section A")                                                       \
+	XX(_index, "{!index:A} emit current record index for section A")
+
+#define SECTION_TEMPLATE_FIELD_MAP(XX)                                                                                 \
+	XX(section, "{$section} Current section index")                                                                    \
+	XX(record, "{$record} Current record index")
+
+/**
+ * @brief Provides enhanced template tag processing for use with a SectionStream
+*/
+class SectionTemplate : public TemplateStream
+{
+public:
+	enum class Command {
+		_unknown = 0,
```

Comment:
Remove leading and trailing underscores where possible.

Reply:
```
 * @note Command tags are prefixed with _ to allow use of reserved words
 * in the Command enumeration. This represents the ! prefix in actual use.
```

Reply:
I could change this to use some other character but as they're all wrapped up in an `enum class` it should be pretty safe.

Reply:
> I could change this to use some other character

Yes, please.